app/components/sidebars/main/channels_list/list/list.js

```diff
@@ -83,18 +86,19 @@ export default class List extends PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
+    static getDerivedStateFromProps(props, state) {
         const {
             canCreatePrivateChannels,
             orderedChannelIds,
             unreadChannelIds,
-        } = this.props;
+        } = state;
 
-        if (nextProps.canCreatePrivateChannels !== canCreatePrivateChannels ||
-            nextProps.unreadChannelIds !== unreadChannelIds ||
-            nextProps.orderedChannelIds !== orderedChannelIds) {
-            this.setState({sections: this.buildSections(nextProps)});
+        if (props.canCreatePrivateChannels !== canCreatePrivateChannels ||
+            props.unreadChannelIds !== unreadChannelIds ||
+            props.orderedChannelIds !== orderedChannelIds) {
+            return {sections: this.buildSections(props)};
```

Comment:
state for this values is never changing so this won't update ever as a consequence. doesn't it make more sense to do this in `componentDidUpdate`